shared/macros-oval.jinja

```diff
@@ -0,0 +1,64 @@
+{{%- macro oval_lineinfile(msg='', path='', parameter='', value='', default='', application='', rpm_package='') -%}}
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    <metadata>
+      <title>{{{ msg or rule_title }}}</title>
+      <affected family="unix">
+        <platform>multi_platform_all</platform>
+      </affected>
+      <description>{{{ rule_title }}}</description>
+    </metadata>
+    <criteria comment="{{{ application }}} is configured correctly or is not installed"
+    operator="OR">
+      <criteria comment="{{{ application }}} is not installed" operator="AND">
+        <extend_definition comment="{{{ application }}} is not required or requirement is unset"
+        definition_ref="{{{ application }}}_not_required_or_unset" />
```

Comment:
These `not_required_or_unset` and its friends are very specific for SSH, and should be part of `oval_sshd_config`.
We should not replicate this mechanism to other sets of rules, they are hard to understand and maintain.

This construct is there to allow a requirement which I'm not sure makes sense.
That the policy gives freedom to have `sshd` installed or not, both situations should be pass.
But if `sshd` is installed, it shall be configured securely.

Reply:
I think in an ideal world, `sshd` installation would be a separate rule which would be co-depended on by other rules. If `sshd` is installed, the profile can select the set of required configuration (no root login, GSSAPI enabled, etc). If it isn't installed, the entire set of rules will pass regardless of configuration state.

IMO, this applies to a lot of our rules for service configuration, since not everyone will run all services. Only a small core of system configuration is truly package-availability independent (`/etc/profile` changes which, while not everyone will want, will always exist). It'd be nice to have a macro-based framework for handling this but I think its tough. You end up needing to pass arrays of identifiers I think...

Anyhow, something to think about.

Reply:
Hi, interesting discussion. This construction has bitten us multiple times, last time in https://github.com/ComplianceAsCode/content/pull/4396. My opinion is that the rules like this should only check the configuration option in the configuration file(s) and they shouldn't check if service is installed or package is installed. Instead, we can do it using a CPE check, which will result to "not applicable" if SSH server isn't available.
The current approach has multiple drawbacks:
* gives a false impression that sshd is configured properly when actually sshd isn't present at all
* the HTML report in OpenSCAP older than 1.3.1 can't display the results when `extend_definition` is used, therefore it's hard to debug why the rules passed or failed
* The checks are not in line with the rule description - the prose doesn't mention installing a package or enabling any service.
*  The rule doesn't check a single isolated configuration item but actually 3 "independent" items, even if it has a single CCE assigned.

I suggest that the new macros generate only the textfilecontent test, without any extend_definition elements  and introduce the CPE applicability check in a separate PR.